CMakeLists.txt

```diff
@@ -1538,8 +927,6 @@ IF(ENABLE_PNETCDF)
   # Check for ncmpi_create in libpnetcdf, define USE_PNETCDF
   # Does the user want to turn on PnetCDF read ability?
   SET(USE_PNETCDF ON CACHE BOOL "")
-  FIND_LIBRARY(PNETCDF NAMES pnetcdf)
-  FIND_PATH(PNETCDF_INCLUDE_DIR pnetcdf.h)
   IF(NOT PNETCDF)
```

Comment:
This now isn't set until the `include(cmake/dependencies.cmake)` line later, so pnetcdf is reported as always disabled (probably requesting a feature and not finding it should be a hard error, which would catch this). I'm not sure if it's as simple as moving that earlier, given there's potentially some interaction between options (another good reason to try and disentangle them all!).

I've whipped up a very simple `FindPNETCDF` module that might be able to sort some of this out. I can make a PR into this branch if you like

Reply:
@ZedThree I've just moved it. I wasn't able to get the tests to pass for the parallel version on my machine but I was getting a different error than the CI anyways. Feel free to make that PR! My goal was to turn as many of the dependencies as possible into `Find*` packages.